### PR TITLE
Search results guidance

### PR DESCRIPTION
**Pull request by allait**
### Search results
image

### No results
image

### Tag search page zendesk tickets with data_request tag

If a support form is filled following a search results "data request" link the resulting Zendesk ticket is tagged with "data_request".

The tag query parameter is checked against a list of allowed tags.

### Checklist

* [ ] Have tests been added to cover any changes?
* [ ] Has the [CHANGELOG](https://github.com/uktrade/data-workspace/blob/master/CHANGELOG.md) been updated?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: dataworkspace/dataworkspace/templates/datasets/index.html [hunk 1]
```diff
@@ -106,6 +107,23 @@ <h2 class="govuk-heading-m">
         {% endif %}
       </ul>
     </nav>
+    {% endif %}
+
+    <div {% if datasets.paginator.count %}class="govuk-!-margin-top-9"{% endif %}>
+      <p class="govuk-body">
+        {% if datasets.paginator.count %}
+          If you haven't found what you're looking for, please:
```
Comment (samuelhwilliams) [inline dataworkspace/dataworkspace/templates/datasets/index.html]:
```suggestion
          If you haven’t found what you’re looking for, please:
```

File: dataworkspace/dataworkspace/tests/datasets/test_views.py [hunk 2]
```diff
@@ -115,6 +124,8 @@ def test_find_datasets_combines_results(client):
         },
     ]
 
+    assert b"If you haven't found what you're looking for" in response.content
```
Comment (samuelhwilliams) [inline dataworkspace/dataworkspace/tests/datasets/test_views.py]:
```suggestion
    assert b"If you haven’t found what you’re looking for" in response.content
```

File: dataworkspace/dataworkspace/apps/core/views.py [hunk 3]
```diff
@@ -53,6 +53,8 @@ class SupportView(FormView):
     form_class = SupportForm
     template_name = 'core/support.html'
 
+    ZENDESK_TAGS = {'data-request': "data_request"}
```
Comment (samuelhwilliams) [inline dataworkspace/dataworkspace/apps/core/views.py]: Different quotes on a single line is a new one ;)
```suggestion
    ZENDESK_TAGS = {"data-request": "data_request"}
```

File: dataworkspace/dataworkspace/templates/datasets/index.html [hunk 4]
```diff
@@ -106,6 +107,23 @@ <h2 class="govuk-heading-m">
...
+        {% else %}
+          There are no results for your search, please:
+        {% endif %}
+      </p>
+      <ul class="govuk-list govuk-list--bullet">
+        <li>check the spelling of your keywords</li>
+        <li>use more general keywords</li>
+        <li>select or deselect different filters</li>
+        <li>request data via our <a href="{% url 'support' %}?tag=data-request">feedback and support form</a></li>
```
Comment (samuelhwilliams) [inline dataworkspace/dataworkspace/templates/datasets/index.html]: We call it "support and feedback" across the rest of the site, so probably good to keep that order the same unless there's a reason to stray.
```suggestion
        <li>request data via our <a href="{% url 'support' %}?tag=data-request">support and feedback form</a></li>
```

File: dataworkspace/dataworkspace/apps/core/views.py (hunk 3 above)
Comment (allait) [reply dataworkspace/dataworkspace/apps/core/views.py]: 😓 maybe we should enable `black` to enforce double quotes

Comment (samuelhwilliams) [reply dataworkspace/dataworkspace/apps/core/views.py]: Fine by me

File: dataworkspace/dataworkspace/tests/datasets/test_views.py [hunk 5]
```diff
@@ -115,6 +124,8 @@ def test_find_datasets_combines_results(client):
         },
     ]
 
+    assert b"If you haven’t found what you’re looking for" in response.content
```
Comment (samuelhwilliams) [inline dataworkspace/dataworkspace/tests/datasets/test_views.py]:
```suggestion
    assert "If you haven’t found what you’re looking for" in response.content.decode(response.charset)
```

Comment (samuelhwilliams) [reply dataworkspace/dataworkspace/tests/datasets/test_views.py]: soz

**Pull request merged**